### PR TITLE
disable neonpixel optimizations on M1 hardware

### PR DESCRIPTION
**Pull request by lstein**
Reinstate neonpixel optimizations with special check for MPS device to avoid (hopefully) performance regression on that platform.

### CODE REVIEW

**Comment by lstein** [comment]
This PR should cure the performance regressions reported by Mac M1 users in issue #364, but this needs confirmation.

**Comment by heurihermilab** [comment]
Yes, this does roll back the performance regression — about a 20x speedup.

Otoh, the returned image is basic color noise no matter how many steps.  After 30 steps:


I suspect that somehow the output of one iteration is not being fed to the next but instead being generated anew...

File: ldm/modules/attention.py [hunk 1]
```diff
@@ -188,9 +190,11 @@ def forward(self, x, context=None, mask=None):
             sim.masked_fill_(~mask, max_neg_value)
             del mask
 
-        # attention, what we cannot get enough of, by halves
-        sim[4:] = sim[4:].softmax(dim=-1)
-        sim[:4] = sim[:4].softmax(dim=-1)
+        if device_type == 'mps':      #special case for M1 - disable neonsecret optimization
+            sim - sim.softmax(dim=-1)
```
Comment (ericmichael) [inline ldm/modules/attention.py:194]: Small typo here appears to be the source of the regression. Should be `sim = sim.softmax(dim=-1)`. After making this change it works on my setup. M1 Mac Mini, macOS Monterey 12.5.1, Conda env `environment-mac.yml`, Python 3.10.5.

**Comment by ericmichael** [comment]
@heurihermilab , can you confirm the fix in https://github.com/lstein/stable-diffusion/pull/414/files#r964373771 works for you?

File: ldm/modules/attention.py (hunk 1 above)
Comment (ericmichael) [reply ldm/modules/attention.py:194]: @heurihermilab, could you confirm this fix works for you?

Comment (heurihermilab) [reply ldm/modules/attention.py:194]: Yes, fixing the - for = typo fixes it.  Thanks!

Comment (i3oc9i) [reply ldm/modules/attention.py:194]: yes performace is back  with a little improvment , too 👍 

50/50 [00:48<00:00,  1.04it/s] neonsecret-optimizations
50/50 [00:53<00:00,  1.06s/it] Release 1.13

Comment (Any-Winter-4079) [reply ldm/modules/attention.py:194]: Yes, this works.

Comment (lstein) [reply ldm/modules/attention.py]: Great, thanks for finding that! I am going to merge into development now.

**Pull request merged**